tests/integration/components/spread-test.js

```diff
@@ -1,30 +1,235 @@
-/* jshint expr:true */
-import { expect } from 'chai'
-import {
-  describeComponent,
-  it
-} from 'ember-mocha'
+import Ember from 'ember'
+import {expect} from 'chai'
+import {$hook, initialize as initializeHook} from 'ember-hook'
+import {describeComponent, it} from 'ember-mocha'
 import hbs from 'htmlbars-inline-precompile'
+import {beforeEach, describe} from 'mocha'
+import sinon from 'sinon'
+import SpreadMixin from 'ember-spread'
 
-describeComponent(
-  'spread',
-  'Integration: SpreadComponent',
-  {
-    integration: true
-  },
-  function () {
-    it('renders', function () {
-      // Set any properties with this.set('myProperty', 'value');
-      // Handle any actions with this.on('myAction', function(val) { ... });
-      // Template block usage:
-      // this.render(hbs`
-      //   {{#spread}}
-      //     template content
-      //   {{/spread}}
-      // `);
-
-      this.render(hbs`{{spread}}`)
-      expect(this.$()).to.have.length(1)
-    })
-  }
-)
+import {integration} from 'dummy/tests/helpers/ember-test-utils/describe-component'
+
+describeComponent(...integration('spread'), function () {
+  let handler
+
+  beforeEach(function () {
+    initializeHook()
+
+    this.registry.register('component:spread-test', Ember.Component.extend(SpreadMixin, {
+
+      // == Properties ============================================================
+
+      hook: 'spread-test',
+      layout: hbs`
+        <div data-test={{hook 'spread-property'}}>
+          {{property}}
+        </div>
+      `,
+
+      // == Actions ===============================================================
+
+      click () {
+        this.onClick()
+      }
+    }))
+
+    handler = sinon.spy()
+  })
+
+  describe('when using the default (options) spread property', function () {
+    beforeEach(function () {
+      this.setProperties({
+        options: {
+          property: 'Neat',
+          onClick: handler
+        }
+      })
+
+      this.render(hbs`
+        {{spread-test
+          options=options
+        }}
+      `)
+    })
+
+    it('should bind spread properties as local properties', function () {
+      expect($hook('spread-property').text().trim()).to.equal('Neat')
+    })
+
+    it('should bind spread functions as local functions', function () {
+      $hook('spread-test').click()
+
+      expect(handler).to.have.callCount(1)
+    })
```

Comment:
Yeah, totally get the hesitation here - technically I'm testing the spread interface, so the real test is the fact the the component implementation is able to use `this.onClick()`

I think I'll separate out the action test so that I can give it a title that is more description of the fact that the beforeEach is very significant